go/feast/repoconfig.go

```diff
@@ -0,0 +1,86 @@
+package feast
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/ghodss/yaml"
+	"os"
+	"path/filepath"
+)
+
+type RepoConfig struct {
+	// Feast project name
+	Project string `json:"project"`
+	// Feast provider name
+	Provider string `json:"provider"`
+	// Path to the registry. Custom registry loaders are not yet supported
+	// Registry string `json:"registry"`
+	Registry interface{} `json:"registry"`
+	// Online store config
+	OnlineStore map[string]interface{} `json:"online_store"`
+	// Offline store config
+	OfflineStore map[string]interface{} `json:"offline_store"`
+	// Feature server config (currently unrelated to Go server)
+	FeatureServer map[string]interface{} `json:"feature_server"`
+	// Feature flags for experimental features
+	Flags map[string]interface{} `json:"flags"`
+}
+
+// NewRepoConfig reads file <repoPath>/feature_store.yaml
+// and converts the YAML format into RepoConfig struct.
+// It first uses repoPath to read feature_store.yaml if it exists,
+// however if it doesn't then it checks configJSON and tries parsing that.
+func NewRepoConfig(repoPath string, configJSON string) (*RepoConfig, error) {
```

Comment:
Would it make sense to have a `NewRepoConfigFromJson` and a wrapping `NewRepoConfigFromFile` (instead of the conditional arguments which is a bit less intuitive)

Reply:
Makes sense, I will separate this init function!